Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/EamDbUtil.java

```diff
@@ -171,27 +171,60 @@ public static boolean schemaVersionIsSet(Connection conn) {
      * upgrade fails, the Central Repository will be disabled and the current
      * settings will be cleared.
      */
-    @Messages({"EamDbUtil.centralRepoUpgradeFailed.message=Failed to upgrade central repository. It has been disabled."})
+    @Messages({"EamDbUtil.centralRepoDisabled.message= The Central Repository has been disabled.",
+        "EamDbUtil.centralRepoUpgradeFailed.message=Failed to upgrade Central Repository.",
+        "EamDbUtil.centralRepoConnectionFailed.message=Unable to connect to Central Repository.",
+        "EamDbUtil.exclusiveLockAquisitionFailure.message=Unable to acquire exclusive lock for Central Repository."})
     public static void upgradeDatabase() throws EamDbException {
         if (!EamDb.isEnabled()) {
             return;
         }
-
+        EamDb db = null;
         CoordinationService.Lock lock = null;
+        String messageForDialog = "";
+        //get connection
         try {
-            EamDb db = EamDb.getInstance();
-
-            // This may return null if locking isn't supported, which is fine. It will
-            // throw an exception if locking is supported but we can't get the lock
-            // (meaning the database is in use by another user)
-            lock = db.getExclusiveMultiUserDbLock();
-
-            db.upgradeSchema();
-
-        } catch (EamDbException | SQLException | IncompatibleCentralRepoException ex) {
-            LOGGER.log(Level.SEVERE, "Error updating central repository", ex);
-
-            // Disable the central repo and clear the current settings.
+            db = EamDb.getInstance();
+        } catch (EamDbException ex) {
+            LOGGER.log(Level.SEVERE, "Error updating central repository, unable to make connection", ex);
+            messageForDialog = Bundle.EamDbUtil_centralRepoConnectionFailed_message() + Bundle.EamDbUtil_centralRepoDisabled_message();
+        }
+        //get lock necessary for upgrade
+        if (db != null) {
+            try {
+                // This may return null if locking isn't supported, which is fine. It will
+                // throw an exception if locking is supported but we can't get the lock
+                // (meaning the database is in use by another user)
+                lock = db.getExclusiveMultiUserDbLock();
+            } catch (EamDbException ex) {
+                LOGGER.log(Level.SEVERE, "Error updating central repository, unable to acquire exclusive lock", ex);
```

Comment:
Should this continue with the upgrade after failing to get the lock?

Reply:
No, good catch we definitely shouldn't in my attempts to remove nested try catch blocks it looks like I didn't do it correct.